rest-server/src/models/user.js

```diff
@@ -37,49 +36,126 @@ const encrypt = (username, password, callback) => {
   }
 };
 
-const defaultValue = {};
-defaultValue[config.lowdbAdmin] = {
-  passwd: encrypt(config.lowdbAdmin, config.lowdbPasswd),
-  admin: true,
-};
-fse.ensureDirSync(path.dirname(config.lowdbFile));
-const adapter = new FileSync(config.lowdbFile, {defaultValue: defaultValue});
-const db = low(adapter);
+const db = dbUtility.getStorageObject('etcd2', {
+  'hosts': etcdConfig.etcdHosts,
+});
 
 const update = (username, password, admin, modify, callback) => {
-  if (typeof modify === 'undefined' || db.has(username).value() !== modify) {
+  if (typeof modify === 'undefined') {
     callback(null, false);
   } else {
-    encrypt(username, password, (err, derivedKey) => {
-      if (err) {
-        callback(err, false);
+    db.has(etcdConfig.userPath(username), (errMsg, res) => {
+      if (res !== modify) {
+        callback(null, false);
       } else {
-        if (modify) {
-          db.set(`${username}.passwd`, derivedKey).write();
-        } else {
-          db.set(username, {passwd: derivedKey, admin: false}).write();
-        }
-        if (typeof admin !== 'undefined') {
-          db.set(`${username}.admin`, admin).write();
-        }
-        callback(null, true);
+        encrypt(username, password, (errMsg, derivedKey) => {
+          if (errMsg) {
+            callback(errMsg, false);
+          } else {
+            if (modify) {
+              db.set(etcdConfig.userPasswdPath(username), derivedKey, (errMsg, res) => {
+                if (errMsg !== null) {
```

Comment:
use `if (errMsg)`? pls refer to [difference between null and undefined](https://stackoverflow.com/a/5101991/7630831)

Reply:
Got it! I will change to if (errMsg). Thanks